scripts/filter_expected_warnings.py

```diff
@@ -55,10 +55,11 @@ def __init__(self, all_lines):
             elif line.startswith('  '):
                 current.otherlines.append(line.strip())
             else:
-                # Assuming all warnings are of the form [path:line: warning:...]
+                # Warnings are usually of the form [path:line: warning:...]
                 # (and the warnings about too many nodes have been filtered out).
-                print('Error filtering warnings: Unexpected input format.')
-                print('  Input:' + line)
```

Comment:
Apologies if I'm missing something here, but it's not clear to me how this changes the exit code of the process, or how this causes travis to mark the job as 'failed' when an unexpected input is received?

Reply:
The error code is set (partly) based on if an unexpected warning is found.  This changes it so that it finds an unexpected 'warning'.